scipy/stats/_stats_py.py

```diff
@@ -3956,7 +3957,84 @@ def __init__(self, msg=None):
         self.args = (msg,)
 
 
-def pearsonr(x, y):
+def _pearsonr_fisher_ci(r, n, confidence_level, alternative):
+    """
+    Compute the confidence interval for Pearson's R.
+
+    Fisher's transformation is used to compute the confidence interval
+    (https://en.wikipedia.org/wiki/Fisher_transformation).
+    """
+    if r == 1:
+        zr = np.inf
+    elif r == -1:
+        zr = -np.inf
+    else:
+        zr = np.arctanh(r)
+    se = np.sqrt(1 / (n - 3))
+
+    if alternative == "two-sided":
+        h = special.ndtri(0.5 + confidence_level/2)
+        zlo = zr - h*se
+        zhi = zr + h*se
+        rlo = np.tanh(zlo)
+        rhi = np.tanh(zhi)
+    elif alternative == "less":
+        h = special.ndtri(confidence_level)
+        zhi = zr + h*se
+        rhi = np.tanh(zhi)
+        rlo = -1.0
+    else:
+        # alternative == "greater":
+        h = special.ndtri(confidence_level)
+        zlo = zr - h*se
+        rlo = np.tanh(zlo)
+        rhi = 1.0
+
+    return ConfidenceInterval(low=rlo, high=rhi)
+
+
+ConfidenceInterval = namedtuple('ConfidenceInterval', ['low', 'high'])
+
+PearsonRResultBase = _make_tuple_bunch('PearsonRResultBase',
+                                       ['statistic', 'pvalue'], [])
+
+
+class PearsonRResult(PearsonRResultBase):
+
+    def __init__(self, statistic, pvalue, alternative, n):
+        super().__init__(statistic, pvalue)
+        self.alternative = alternative
+        self.n = n
```

Comment:
In the original PR, `alternative` and `n` were part of the `extra_field_names` in `_make_tuple_bunch`. This seemed unusual - especially `n`. If these are considered essential parts of the result object, why stop there (e.g. why is `n` so important but not all the data)? But really, why include them at all?

My guess is that the real reason they were included was not to be accessible to the user but just because they were needed by `confidence_interval`. Unfortunately, there was no way to make them private attributes of the result object because `__setattr__` was disabled by `_make_tuple_bunch`. It seems too restrictive to have result objects with methods but no private attributes seems too restrictive, so I modified the `_make_tuple_bunch` code to allow `__setattr__` for now. 

I would prefer to mark these as private (i.e. `_alternative` and `_n`), but for some reason `_make_tuple_bunch` also doesn't want attribute names to begin with an underscore, so I compromised.